### ISSUE TITLE
Proposal: include `lang` rule in `jsx-a11y/strict` config

### ISSUE THREAD

**Issue by rkpatel7** [jsx-eslint/eslint-plugin-jsx-a11y#750]
### Current state
[`jsx-a11y/strict`][1] does not include the [`jsx-a11y/lang`][2] rule but does include the [`jsx-a11y/html-has-lang`][3] rule. 

### Context

Looking at the [`jsx-a11y/lang`][4] source code and [`jsx-a11y/html-has-lang`][5] source code, it appears that the latter ensures that `html` elements have a `lang` attribute with _any_ value and the former additionally checks if the `lang` attribute's value conforms to the [BCP 47][6] list of acceptable tags. To satisfy [WCAG 3.1.1][7], technique [H57][8] is sufficient but that technique requires both of the following to be true:
* `html` element has a `lang` attribute (checked by rule `jsx-a11y/html-has-lang`, included in `jsx-a11y/strict`)
* `lang` attribute conforms to BCP 47 (checked by rule `jsx-a11y/lang`, NOT included in `jsx-a11y/strict`)

### Proposal:
In order to wholly satisfy WCAG 3.1.1:
* `jsx-a11y/lang` MUST be included in `jsx-a11y/strict`
* `jsx-a11y/lang` SHOULD be included `jsx-a11y/recommended`<sup>[1](#footnote)</sup>

----
<a name="footnote">1</a>: I am not sure what the criteria is for including a rule in `jsx-a11y/recommended` but given that WCAG 3.1.1 would only be sufficiently covered with the inclusion of `jsx-a11y/lang`, I think it would make sense to include the rule in both configs.

[1]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/index.js#L200
[2]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/lang.md
[3]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/html-has-lang.md
[4]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/rules/lang.js
[5]: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/src/rules/html-has-lang.js
[6]: https://www.rfc-editor.org/rfc/bcp/bcp47.txt
[7]: https://www.w3.org/WAI/WCAG21/Understanding/language-of-page
[8]: https://www.w3.org/WAI/WCAG21/Techniques/html/H57

**Comment by ljharb** [comment]
Adding rules to a config is a breaking change, and it's not worth making a breaking change just to do that.

A TODO comment, however, to make these changes would be great, for whenever a breaking change is already forced.

**Comment by rkpatel7** [comment]
I think that's a good middle ground. I'd be happy to make a pull request with the TODO comment. Let me know if that is acceptable and, if so, what contributing guidelines/docs there are (I couldn't find any in the repo). Thanks!

**Comment by ljharb** [comment]
Sure, that'd be great. We don't have any docs around contributing; infer from the repo what you can, and I'll provide feedback on the PR as needed.
